internal/pkg/cli/run_local.go

```diff
@@ -815,12 +831,48 @@ func (h *hostDiscoverer) Hosts(ctx context.Context) ([]orchestrator.Host, error)
 			for _, alias := range sc.ClientAliases {
 				hosts = append(hosts, orchestrator.Host{
 					Name: aws.StringValue(alias.DnsName),
-					Port: strconv.Itoa(int(aws.Int64Value(alias.Port))),
+					Port: uint16(aws.Int64Value(alias.Port)),
 				})
 			}
 		}
 	}
 
+	// get rds instances
+	dbs, err := h.rg.GetResourcesByTags(resourcegroups.ResourceTypeRDSInstance, map[string]string{
+		deploy.AppTagKey: h.app,
+		deploy.EnvTagKey: h.env,
+	})
+	switch {
+	case err != nil:
+		return nil, fmt.Errorf("get tagged rds instances: %w", err)
+	case len(dbs) == 0:
+		return hosts, nil
+	}
+
+	filter := &rds.Filter{
+		Name: aws.String("db-instance-id"),
+	}
+	for i := range dbs {
+		filter.Values = append(filter.Values, aws.String(dbs[i].ARN))
+	}
+
+	err = h.rds.DescribeDBInstancesPagesWithContext(ctx, &rds.DescribeDBInstancesInput{
+		Filters: []*rds.Filter{filter},
+	}, func(out *rds.DescribeDBInstancesOutput, lastPage bool) bool {
+		for _, db := range out.DBInstances {
+			if db.Endpoint != nil {
+				hosts = append(hosts, orchestrator.Host{
+					Name: aws.StringValue(db.Endpoint.Address),
+					Port: uint16(aws.Int64Value(db.Endpoint.Port)),
+				})
+			}
+		}
+		return true
+	})
```

Comment:
I don't see where the pages are used at all. Can we call `DescribeDBInstancesWithContext` instead?

Reply:
the pages happen automagically by using these paginator functions! basically, the function we've passed to the function will be called `n` times, based on how many pages of the response there are. it basically wraps this loop for us: https://github.com/aws/copilot-cli/blob/mainline/internal/pkg/aws/resourcegroups/resourcegroups.go#L58-L86